build_runner/test/server/build_updates_client/reloading_manager_test.dart

```diff
@@ -30,7 +30,7 @@ void main() {
           methods.reloadModule,
           methods.getModuleLibraries,
           methods.reloadPage,
-          (id) => moduleParentsGraph[id],
+          (id) => moduleParentsGraph[id] ?? const [],
```

Comment:
In `graphs` 1.x, the `edges`  function in `stronglyConnectedComponents` is not allowed to return `null` anymore. That sounds reasonable to me, but it's tricky to catch in non-migrated packages.

I don't know if `build_runner/web/hot_reload_client.dart` needs changes too, is there an integration test to find out?

Reply:
I don't think we have a hot reload test any longer in build_runner (I don't think we do any webdriver testing which this would require). Webdev is generally where tests like that live but it has its own hot module reloading script which is a bit different if I recall correctly.

I believe it wouldn't return null though, all nodes in the graph should have parents, or else they wouldn't be in the app at all. That assumes we do skip doing anything if you edit a module not depended on by the app.

Reply:
Honestly we could just finally remove entirely the ability to hot reload from this package, we really expect people to go through webdev for that now and the implementation in this package is likely to bitrot.